app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt

```diff
@@ -1114,6 +997,126 @@ private fun sponsoredShortcut(sponsoredShortcutTitle: String) =
 private fun storyByTopicItem(composeTestRule: ComposeTestRule, position: Int) =
     composeTestRule.onNodeWithTag("pocket.categories").onChildAt(position - 1)
 
+private fun appItemWithResourceId(resourceId: String) =
+    mDevice.findObject(UiSelector().resourceId(resourceId))
+
+private fun appItemContainingText(itemText: String) =
+    mDevice.findObject(UiSelector().textContains(itemText))
+
+private fun appItemStateWithResourceId(resourceId: String, state: Boolean) =
+    mDevice.findObject(UiSelector().resourceId(resourceId).checked(state))
+
+private fun appItemWithResourceIdAndDescription(resourceId: String, description: String) =
+    mDevice.findObject(UiSelector().resourceId(resourceId).descriptionContains(description))
+
+private fun appItemWithResourceIdAndText(resourceId: String, text: String) =
+    mDevice.findObject(UiSelector().resourceId(resourceId).text(text))
+
+private fun assertAppItemsWithResourceId(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
+
+private fun assertAppItemsContainingText(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
+
+private fun assertAppItemsStateWithResourceId(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
+
+private fun assertAppItemsWithResourceIdAndDescription(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
+
+private fun assertAppItemsWithResourceIdAndText(vararg appItems: UiObject) {
+    for (appItem in appItems) {
+        assertTrue(appItem.waitForExists(waitingTime))
+    }
+}
```

Comment:
Looks awesome. I wonder if these should be moved to a MatcherHelper class or something of this sort, to be re-used in all robots. @AaronMT ?

Reply:
I'm OK with doing this outside this ticket, file a Jira.